src/sentry/issues/status_change_message.py

```diff
@@ -18,6 +20,7 @@ class StatusChangeMessage:
     project_id: int
     new_status: int
     new_substatus: int | None
+    id: str = uuid4()
```

Comment:
This will result in all messages having the same uuid.

```
>>> abc = StatusChangeMessage(['12'], 1, 0, None)
>>> bcd = StatusChangeMessage(['13'], 1, 0, None)
>>> abc
StatusChangeMessage(fingerprint=['12'], project_id=1, new_status=0, new_substatus=None, id=UUID('d95199c9-95cb-4be7-8b66-6e4b50a9ffb4'))
>>> bcd
StatusChangeMessage(fingerprint=['13'], project_id=1, new_status=0, new_substatus=None, id=UUID('d95199c9-95cb-4be7-8b66-6e4b50a9ffb4'))
```

`id: UUID = field(default_factory=uuid4)`
or
`id: str = field(default_factory=lambda: uuid4().hex)` should do what you want

Reply:
@wedamija yea, let me add a test for that